api/v1alpha1/api_key_auth_types.go

```diff
@@ -22,6 +22,18 @@ type APIKeyAuth struct {
 	// ExtractFrom is where to fetch the key from the coming request.
 	// The value from the first source that has a key will be used.
 	ExtractFrom []*ExtractFrom `json:"extractFrom"`
+
+	// ForwardClientIDHeader is the name of the header to forward the client identity to the backend
+	// service. The header will be added to the request with the client id as the value.
+	//
+	// +optional
+	ForwardClientIDHeader *string `json:"forwardClientIDHeader,omitempty"`
+
+	// SanitizeAPIKey indicates whether to remove the API key from the request
+	// before forwarding it to the backend service.
+	//
+	// +optional
+	SanitizeAPIKey *bool `json:"sanitizeAPIKey,omitempty"`
```

Comment:
any prior art in this area ?
hard to make a choice b/w this and `SuppressCredentials`

Reply:
sure Sanitize works, is `apiKeyAuth.sanitizeAPIKey` too wordy ? should it be `apiKeyAuth.sanitize` instead ?